src/libs/actions/PersonalDetails.js

```diff
@@ -67,6 +67,10 @@ function getDisplayName(login, personalDetail) {
         return userLogin;
     }
 
+    if (userDetails.displayName) {
```

Comment:
This was removed [here](https://github.com/Expensify/Expensify.cash/pull/2204) due to [this reason](https://github.com/Expensify/Expensify.cash/pull/2204/files#r605930868), so re-adding it may re-introduce an issue.

Reply:
Maybe instead we can check if `(`${firstName} ${lastName}`).trim() ` is equal to `userDetails.displayName` and if not update it appropriately?

Reply:
Good catch! It looks like that issue was caused by https://github.com/Expensify/Expensify.cash/blob/29d650ba21bf9e02b6de5e590caa3e738605282a/src/libs/actions/PersonalDetails.js#L220, since we were checking the displayName first. I think if we change this logic to return `firstName lastName` first, and fallback on displayName, this should fix the original issue and prevent the regression.

I'll do some testing and add a test case to the PR tests to ensure if I change the first/last name we don't see that issue happen.

Reply:
I did some testing and I just want to confirm the behavior mentioned in [this comment](https://github.com/Expensify/Expensify.cash/pull/2204/files#r605930868) is fixed. @NikkiWines do the testing steps below sound correct for the issue described, or was the issue something different? cc @marcaaron for any additional context

On master
0. User A types, User B sees email instead of personalDetail name
1. User A changes name, User A sees personalDetail name updated everywhere
2. User A types, User B sees old name in "is typing" indicator
3. User B refreshes
4. User A types after refresh, "Is typing" indicator shows User A's old name

On this branch w/ latest update
0. User A types, User B sees personalDetail name
1. User A changes name, User A sees personalDetail name updated everywhere
2. User A types, User B sees old name in "is typing" indicator
3. User B refreshes
4. User A types after refresh, "Is typing" indicator shows User A's new name

Reply:
That sounds right to me. The other user should not expect any realtime update from a name change (unless they are changing their own name).